File: roman.cpp
#include <iostream>

std::string romanUnit(int num) {
        std::string roman = "";
        if (num < 4) {
            for (size_t i = 0; i < num; i++) {
                roman += "I";
            }
        } else if (num == 4) {
            roman += "IV";
        } else if (num == 5) {
            roman += "V";
        } else if (num == 9) {
            roman += "IX";
        } else {
            roman += "V";
            for(size_t i = 0; i < num - 5; i++) {
                roman += "I";
            }
        }
        return roman;
}
    
std::string romanTens(int num) {
        std::string roman = "";
        if (num < 4) {
            for (size_t i = 0; i < num; i++) {
                roman += "X";
            }
        } else if (num == 4) {
            roman += "XL";
        } else if (num == 5) {
            roman += "L";
        } else if (num == 9) {
            roman += "XC";
        } else {
            roman += "L";
            for(size_t i = 0; i < num - 5; i++) {
                roman += "X";
            }
        }
        return roman;
}
    
std::string romanHundreds(int num) {
        std::string roman = "";
        if (num < 4) {
            for (size_t i = 0; i < num; i++) {
                roman += "C";
            }
        } else if (num == 4) {
            roman += "CD";
        } else if (num == 5) {
            roman += "D";
        } else if (num == 9) {
            roman += "CM";
        } else {
            roman += "D";
            for(size_t i = 0; i < num - 5; i++) {
                roman += "C";
            }
        }
        return roman;
}
    
std::string romanThousands(int num) {
        std::string roman = "";
        for(size_t i = 0; i < num; i++) {
            roman += "M";
        }
        return roman;
}
    
std::string intToRoman(int num) {
        if (num != 0) {
                int numThousands = num/1000;
                int numHundreds = (num - numThousands*1000)/100;
                int numTens = (num - numThousands*1000 - numHundreds*100)/10;
                int numUnit = num - numThousands*1000 - numHundreds*100 - numTens*10;
                return (romanThousands(numThousands) + romanHundreds(numHundreds) + romanTens(numTens) + romanUnit(numUnit));
        } else return "Error";
}

int romanToInt(std::string s) {
        int finalNumber = 0;
        
        for (size_t i = 0; i < s.size(); i++) {
            switch(s[i]) {
                case 'I':
                    if (s[i+1] == 'V' ) {
                        finalNumber += 4;
                        i++;
                    } else if (s[i+1] == 'X') {
                        finalNumber += 9;
                        i++;
                    } else finalNumber++;
                    break;
                    
                case 'V': finalNumber += 5;
                    break;
                
                case 'X': 
                    if (s[i+1] == 'L' ) {
                        finalNumber += 40;
                        i++;
                    } else if (s[i+1] == 'C') {
                        finalNumber += 90;
                        i++;
                    } else finalNumber += 10;
                    break;
                    
                case 'L': finalNumber += 50;
                    break;
                    
                case 'C':
                    if(s[i+1] == 'D') {
                        finalNumber += 400;
                        i++;
                    } else if(s[i+1] == 'M') {
                        finalNumber += 900;
                        i++;
                    } else finalNumber += 100;
                    break;
                
                case 'D': finalNumber += 500;
                    break;
                
                case 'M': finalNumber += 1000;
                    break;
            }
        }
        return finalNumber;
}

int main() {
        bool useAgain = true;
        while (useAgain)
        {
                std::cout << std::endl << "What conversion do you want to make?" << std::endl
                << "1- Integer to Roman" << std::endl
                << "2- Roman to Integer" << std::endl
                << "3- Quit" << std::endl;

                int option;
                std::cin >> option;

                int integer;
                std::string roman;
                switch (option)
                {
                        case 1:
                                std::cout << std::endl << "Enter your integer: ";
                                std::cin >> integer;
                                std::cout << intToRoman(integer) << std::endl;
                                break;
                        case 2:
                                std::cout << std::endl << "Enter your roman: ";      
                                std::cin >> roman;
                                std::cout << romanToInt(roman) << std::endl;
                                break;
                        case 3: useAgain = false;
                                break;
                }
        }
}

File: roman.py
def romanUnit(num):
    roman = ""
    if num < 4:
        for x in range(0, num):
            roman += "I"
    elif num == 4:
        roman += "IV"
    elif num == 5:
        roman += "V"
    elif num == 9:
        roman += "IX"
    else:
        roman += "V"
        for x in range(0, num - 5):
            roman += "I"
    return roman


def romanTens(num):
    roman = ""
    if num < 4:
        for x in range(0, num):
            roman += "X"
    elif num == 4:
        roman += "XL"
    elif num == 5:
        roman += "L"
    elif num == 9:
        roman += "XC"
    else:
        roman += "L"
        for x in range(0, num - 5):
            roman += "X"
    return roman


def romanHundreds(num):
    roman = ""
    if num < 4:
        for x in range(0, num):
            roman += "C"
    elif num == 4:
        roman += "CD"
    elif num == 5:
        roman += "D"
    elif num == 9:
        roman += "CM"
    else:
        roman += "D"
        for x in range(0, num - 5):
            roman += "C"
    return roman


def romanThousands(num):
    roman = ""
    for x in range(0, num):
        roman += "M"
    return roman


def intToRoman(num):
    if num != 0:
        numThousands = int(num/1000)
        numHundreds = int((num - numThousands*1000)/100)
        numTens = int((num - numThousands*1000 - numHundreds*100)/10)
        numUnit = int(num - numThousands*1000 - numHundreds*100 - numTens*10)
        return romanThousands(numThousands) + romanHundreds(numHundreds) + romanTens(numTens) + romanUnit(numUnit)
    else:
        return "Error"


def romanToInt(s):
    finalNumber = 0
    i = 0

    while i < len(s):
        if s[i] == 'I':
            if i+1 < len(s):
                if s[i+1] == 'V':
                    finalNumber += 4
                    i += 1
                elif (s[i+1] == 'X'):
                    finalNumber += 9
                    i += 1
                else:
                    finalNumber += 1
            else:
                finalNumber += 1

        elif s[i] == 'V':
            finalNumber += 5

        elif s[i] == 'X':
            if i+1 < len(s):
                if s[i+1] == 'L':
                    finalNumber += 40
                    i += 1
                elif s[i+1] == 'C':
                    finalNumber += 90
                    i += 1
                else:
                    finalNumber += 10
            else:
                finalNumber += 10

        elif s[i] == 'L':
            finalNumber += 50

        elif s[i] == 'C':
            if i+1 < len(s):
                if s[i+1] == 'D':
                    finalNumber += 400
                    i += 1
                elif s[i+1] == 'M':
                    finalNumber += 900
                    i += 1
                else:
                    finalNumber += 100
            else:
                finalNumber += 100

        elif s[i] == 'D':
            finalNumber += 500

        elif s[i] == 'M':
            finalNumber += 1000
        i += 1
    return finalNumber


useAgain = True
while useAgain:
    print("\n" + "What conversion do you want to make?" + "\n")
    print("1- Integer to Roman")
    print("2- Roman to Integer")
    print("3- Quit")

    option = int(input())

    if option == 1:
        print("\n" + "Enter your integer: ")
        integer = float(input())
        print(intToRoman(integer) + "\n")
    if option == 2:
        print("\n" + "Enter your roman: ")
        roman = str(input())
        print(str(romanToInt(roman)) + "\n")
    if option == 3:
        useAgain = False
